### PR TITLE
[1.x] Backporting TensorRT-Gluon Partition API (and TensorRT 7 support)

### PR DESCRIPTION
**Pull request by Kh4L**
## Description ##
Backported PRs (some of them are slightly modified to be compatible with v1.x):

- Change include to relative in nvvm_to_onnx.cc #18249 
- Drop list support in optimize_for #18483
- Add better partial args/aux handling in symbol optimize_for #18350
- Partition API adding and deleting new params to Block and Symbol #18405
- Remove check for subgraph with cycles #18555
- MXNet-TRT: Add PrePartition param caching - move init_tensorrt_params logic #18490 
- Fix a bug in TensorRT partitioning #18870

### CODE REVIEW

**Comment by Kh4L** [comment]
@mxnet-bot run ci [edge, unix-cpu]

**Comment by Kh4L** [comment]
@mxnet-bot run ci [unix-cpu, unix-gpu]

**Comment by Kh4L** [comment]
@mxnet-bot run ci [unix-gpu]

File: ci/jenkins/Jenkins_steps.groovy [hunk 1]
```diff
@@ -896,24 +896,6 @@ def test_unix_python3_mkldnn_nocudnn_gpu() {
     }]
 }
 
-def test_unix_python3_tensorrt_gpu() {
```
Comment (samskalicky) [inline ci/jenkins/Jenkins_steps.groovy:899]: @mseth10 I guess this is the cause of the CI job issue

Comment (samskalicky) [reply ci/jenkins/Jenkins_steps.groovy:899]: @Kh4L you'd need to get rid of this too if you wanna remove this test:
https://github.com/apache/incubator-mxnet/blob/9981e847fff9270268385068c5b7d0c3929e46f9/ci/jenkins/Jenkinsfile_unix_gpu#L53

Comment (mseth10) [reply ci/jenkins/Jenkins_steps.groovy:899]: that's right, if we remove this test we should also remove it from jenkins pipeline

Comment (mseth10) [reply ci/jenkins/Jenkins_steps.groovy:899]: in master, it's removed in this commit
https://github.com/apache/incubator-mxnet/commit/c1098aa33d6795f84a19601d0319d5bb8e19f317#diff-0c2b22569b87fa79153336ea3ac94f33

Comment (Kh4L) [reply ci/jenkins/Jenkins_steps.groovy:899]: Thanks for investigating, I missed that file...

**Pull request merged**